### ISSUE TITLE
Change labels of NS sensor on charts

### ISSUE THREAD

**Issue by dharezlak** [ismop/dap#119]
Change labels of NS sensors on charts to more meaningful, such that allow identification of the sensors' locations. For example "neosentio.3_26_0_4 (Temperatura)". However, Daniel Kessler said that these IDs may change.

**Comment by dharezlak** [comment]
UI uses `parameter.custom_id (parameter.measurement_type_name)` to generate chart labels.

**Comment by dharezlak** [comment]
Proposal:

* pump_monitoring dla PV-> Poziom wody - czujnik 1
* pump_monitoring dla 1_PV-> Poziom wody - czujnik 2
* pump_monitoring dla 2_PV-> Poziom wody - średnia dla czujników 1 i 2
* pump_monitoring dla SP-> Poziom wody - wartość zadana

**Comment by nowakowski** [comment]
@dharezlak Could you suggest a suitable label (by providing examples)?

**Comment by dharezlak** [comment]
This issue has been originally reported by @balis with https://github.com/ismop/eitwid/issues/105. As it was a DAP-only issue I moved it here. I suppose @balis should have more details on it.

**Comment by nowakowski** [comment]
@balis We'll need a more detailed explanation of what kinds of labels are required. Could you give us some examples?

**Comment by mkasztelnik** [comment]
I propose to treat this issue as a possibility to fix all naming problems we currently have in DAP. In device, parameter we have following values:
  * `Device.name`
  * `Parameter.name`
  * `Parameter.measurement_type_unit`
  * `Parameter.measurement_type_name`

(I know that `measurement_type_*` are in different model but from serialization point of view these values are serialized as parameter values).

Currently in UI we need to cheat and use something like:
```java
// X axis name - this is not OK
chartSeries.setName(device.getCustomId() + " (" + parameter.getMeasurementTypeName() + ")");
// Y axis name - this is fine
parameter.getMeasurementTypeName() + " [" + parameter.getMeasurementTypeUnit() + "]"
```

As you can see we use `getMeasurementTypeName()` to present something meaningful to the user, but instead we should be using:

```java
// e.g. UT2 (Temperature)
chartSeries.setName(device.getCustomId() + " (" + parameter.getName() + ")");
// Y axis name - this is fine
parameter.getMeasurementTypeName() + " [" + parameter.getMeasurementTypeUnit() + "]"
```

Now we cannot do this, because there is a lot of "unknown parameter" parameter names. It would be also nice to have meaningful device name (e.g. `Światłowód 245m`) instead of custom id which can looks as follow: `ASP.POMIARY_GESO_1000.F_0245`

**Comment by nowakowski** [comment]
@mkasztelnik I agree that the values of `Parameter.parameter_name` need some massaging since right now it mostly duplicates `Parameter.custom_id` (or else is set to `unknown parameter` by the parameter discovery service). I could write a script to convert these values to something useful (e.g. "Światłowód 254m"). Should I?

Regarding `Device`, I don't want to reuse `custom_id` as a human-readable label - it was not meant for that purpose. Instead, I can add a separate field (e.g. `device_name`) and initialize it to whatever you want (e.g. "Czujnik temperatury" or similar).

**Comment by mkasztelnik** [comment]
@nowakowski it would be nice to make it right :smile: As I was proposing above it would be nice to have unified axis name formatting in UI for all series like:

```java
// e.g. UT2 (Temperature)
chartSeries.setName(device.getName() + " (" + parameter.getName() + ")");
// Y axis name - this is fine
parameter.getMeasurementTypeName() + " [" + parameter.getMeasurementTypeUnit() + "]"
```
To make it possible correct data in DAP is needed. 

According to device name I meant exactly how you describe it. As you can see I already changed `device.getCustomId()` into `Device.getName()` in my example.

**Comment by mkasztelnik** [comment]
@nowakowski any update of this issue? This is quite urgent for implementing https://github.com/ismop/eitwid/issues/136

**Comment by nowakowski** [comment]
@mkasztelnik Yeah, there is progress. While I'm not sure if we can close this issue altogether, I have worked with Daniel Kessler to re-register NS sensors and it seems he's finally happy with the result:

<pre>
Piotrze
O to chodziło. Teraz mogę wybierać co mnie interesuje. Super.
</pre>

Note, however, that Daniel has also asked me to provide wildcard support in the Device filter (Analysis view) so he could input something like "Neosentio_3_*_1_4". I told him it was a GUI issue and suggested filing an eitwid ticket.

**Comment by dharezlak** [comment]
Can we move this issue forward? Things to do:
* [ ] add a `name` field to the Device entity and initialize it with the current value of the `custom_id` field.

For #90 we could do the following:
* [ ] set `Device.name` to `Poziom wody`,
* [ ] respectively set `Parameter.parameter_name` to `poziom dla czujnika 1`, `poziom dla czujnika 2`, `poziom zadany`, `średni poziom dla czujników 1 i 2`, `zużycie energii pomp`, `aktywność pompy 1` oraz `aktywność pompy 2`.

**Comment by nowakowski** [comment]
@dharezlak Can do.

**Comment by mkasztelnik** [comment]
Device prefixes:
  - Fiber optic: FO
  - Budokop: B
  - Neosentio: N

Measurement type prefix:
  - U - czujniki cisnień porowych
  - T - temperatura
  - SV - naprężenie
